### ISSUE TITLE
problem with xgb.predict

### ISSUE THREAD

**Issue by teopapad92** [dmlc/xgboost#3251]
i would like to ask a question about a problem that i have for the last couple days.
First of all i am a beginner in machine learning and this is my first time using the XGBoost algorithm so excuse me for any mistakes I have done.

I trained my model to predict whether a log file is malicious or not. After i save and reload my model on a different session i use the predict function which seems to be working normally ( with a few deviations in probabilities but that is another topic, I know I, have seen it in another topic )

The problem is this: Sometimes when i try to predict a "small" csv file after load it seems to be broken predicting only the Zero label, even for indexes that are categorized correct previously.
For example, i load a dataset containing 20.000 values , the predict() is working. I keep only the first 5 of these values using pandas drop, again its working. If i save the 5 values on a different csv and reload it its not working. The same error happens if i just remove by hand all indexes (19.995) and save file only with 5 remaining.

I would bet it is a size of file problem but when i drop the indexes on the dataframe through pandas it seems to be working
Also the number 5 ( of indexes ) is for example purpose the same happens if I delete a large portion of the dataset.
I first came up with this problem after trying to verify by hand some completely new logs, which seem to be classified correctly if thrown into the big csv file but not in a new file on their own.

##Here is my load and predict code

    ##IMPORTS
    
    df = pd.read_csv('big_test.csv')
    
    df3 = pd.read_csv('small_test.csv')
    
    #This one is necessary for the loaded_model
    class ColumnSelector(BaseEstimator, TransformerMixin):
    def init(self, column_list):
    self.column_list = column_list
    
    def fit(self, x, y=None):
        return self
    
    def transform(self, x):
        if len(self.column_list) == 1:
            return x[self.column_list[0]].values
        else:
            return x[self.column_list].to_dict(orient='records')
    
    loaded_model = joblib.load('finalized_model.sav')
    
    result = loaded_model.predict(df)
    print(result)
    
    df2=df[:5]
    result2 = loaded_model.predict(df2)
    print(result2)
    
    result3 = loaded_model.predict(df3)
    print(result3)

The results i get are these:

[1 0 1 ... 0 0 0]

[1 0 1 0 1]

[0 0 0 0 0]

I can provide any code even from training or my dataset if necessary.

*EDIT: I use a pipeline for my data. I tried to reproduce the error after using xgb to fit the iris data and i could not. Maybe there is something wrong with my pipeline? the code is below :

    df = pd.read_csv('big_test.csv')
    # df.info()
    
    
    # Split Dataset
    attributes = ['uri','code','r_size','DT_sec','Method','http_version','PenTool','has_referer', 'Lang','LangProb','GibberFlag' ]
    x_train, x_test, y_train, y_test = train_test_split(df[attributes], df['Scan'], test_size=0.2, 
    stratify=df['Scan'], random_state=0)
    
    x_train, x_dev, y_train, y_dev = train_test_split(x_train, y_train, test_size=0.2, 
                                                      stratify=y_train, random_state=0)
    # print('Train:', len(y_train), 'Dev:', len(y_dev), 'Test:', len(y_test))
    
    # set up graph function
    def plot_precision_recall_curve(y_true, y_pred_scores):
        precision, recall, thresholds = precision_recall_curve(y_true, y_pred_scores)
        return ggplot(aes(x='recall', y='precision'), 
                      data=pd.DataFrame({"precision": precision, "recall": recall})) + geom_line()
    
    
    # XGBClassifier
    
    class ColumnSelector(BaseEstimator, TransformerMixin):
        def __init__(self, column_list):
            self.column_list = column_list
            
        def fit(self, x, y=None):
            return self
        
        def transform(self, x):
            if len(self.column_list) == 1:
                return x[self.column_list[0]].values
            else:
                return x[self.column_list].to_dict(orient='records')
    
    
    count_vectorizer = CountVectorizer(analyzer='char', ngram_range=(1, 2), min_df=10)
    dict_vectorizer = DictVectorizer()
    xgb = XGBClassifier(seed=0)
    
    pipeline = Pipeline([
        ("feature_union", FeatureUnion([
            ('text_features', Pipeline([
                ('selector', ColumnSelector(['uri'])), 
                ('count_vectorizer', count_vectorizer)
            ])),
            ('categorical_features', Pipeline([
                ('selector', ColumnSelector(['code','r_size','DT_sec','Method','http_version','PenTool','has_referer', 'Lang','LangProb','GibberFlag' ])),
                ('dict_vectorizer', dict_vectorizer)
            ]))
        ])),
        ('xgb', xgb)
    ])
    
    pipeline.fit(x_train, y_train)

    filename = 'finalized_model.sav'
    joblib.dump(pipeline, filename)

**Comment by tqchen** [comment]
general questions are now moved to https://discuss.xgboost.ai/.
